### ISSUE TITLE
Save “smFRET-nsALEX” (PIE) measurement to Photon-HDF5

### ISSUE THREAD

**Issue by ncodina** [Photon-HDF5/phconvert#22]
Hi!

We want to save a “smFRET-nsALEX” (PIE) measurement to Photon-HDF5 and we have questions about the alternation parameters. 

We have .PTU files from PicoQuant, type rtHydraHarp2T3. The function `pqreader.load_ptu` (https://github.com/Photon-HDF5/phconvert/blob/master/phconvert/pqreader.py) decodes the files.

To write the Photon-HDF5 file we used:

```python
measurement_specs = dict(
    measurement_type = 'smFRET-nsALEX',
    laser_repetition_rate = 20000000.0, # excitation laser pulse repetition rate in Hertz (1/5e-08)
    detectors_specs = {'spectral_ch1': [1],  # list of donor's detector IDs
                       'spectral_ch2': [0]})  # list of acceptor's detector IDs
```

```python
photon_data = dict(
    timestamps=timestamps,
    detectors=detectors,
    timestamps_specs={'timestamps_unit': 1e-07},
    nanotimes=nanotimes,
    nanotimes_specs={'tcspc_unit': 1.6e-11,'tcspc_num_bins': 6250.0,'tcspc_range':6250.0*1.6e-11},
    measurement_specs = measurement_specs)
```
using:
timestamps_unit: "MeasDesc_GlobalResolution"
tcspc_unit: "MeasDesc_Resolution"

```python
setup = dict(
    num_pixels = 2,               # using 2 detectors
    num_spots = 1,                # a single confocal excitation
    num_spectral_ch = 2,          # donor and acceptor detection 
    num_polarization_ch = 1,      # number of detected polarization states (none = 1 and 2 polarizations = 2)
    num_split_ch = 1,             # no beam splitter
    modulated_excitation = True, # CW excitation, no modulation - True if excitation is alternated
    lifetime = True)             # no TCSPC in detection
```

However, when we load the .h5 file generated onto FretBursts a warning message appears saying: 
WARNING: No alternation defintion found.
and 'Data' object has no attribute 'A_em'

Are there other alternation parameters that we should specify?

**Comment by tritemio** [comment]
Seems ok. Moreover, if phconvert saves the file it means that it passes all the validation tests. You can also add the **optional fields** `alex_excitation_period1` and `alex_excitation_period2` which contain ranges of TCSPC histograms for the donor and acceptor periods (`spectra_ch1` and `spectral_ch2` respectively). These fields are just a  convenience for analysis programs as they allow to start with a pre-set selection of donor and acceptor ranges in the the TCSPC histogram.

The WARNING in FRETBursts is benign but you should follow the procedure illustrated below:

1. declare the donor and acceptor ranges in the TCPC histogram:

```
d.add(A_ON=(200, 1500), D_ON=(1750, 3200))
```
2. Visually check the selection with:

```
bpl.plot_alternation_hist(d)
```

3. Once the ranges are satisfactory, apply them with:

```
loader.alex_apply_period(d)
```

Then you can proceed to background estimation and burst search. Note that the attribute `A_em` in `Data` is only created after step 3. This procedure is illustrated in the nsALEX notebook https://github.com/tritemio/FRETBursts/blob/master/notebooks/FRETBursts%20-%20ns-ALEX%20example.ipynb. 

Is this what you are doing?

**Comment by ncodina** [comment]
Thanks so much!

No, I tried but I didn't know where the values in `d.add(A_ON=(,), D_ON=(,))` came from. 
I've found that the units are TDC or TAC bin unit, and that we can get them using `tcspc_unit` (in my case, 1.6e-11):
5e-08/1.6e-11 = 3125
1e-07/1.6e-11 = 6250

Could you let me know what should we take into account when selecting these periods?
I'm attaching an example of my file where I used `d.add(D_ON = (50, 2500), A_ON = (3175, 5650))`.
<img width="579" alt="screen shot 2017-01-17 at 16 22 17" src="https://cloud.githubusercontent.com/assets/23275480/22033146/4663beb6-dcdf-11e6-8305-1e87224357aa.png">

**Comment by tritemio** [comment]
These selections look good. Depending on which kind of fitting you want to do on the lifetime, it may be useful to include a bit of the baseline before the peak (in the screenshot there is some baseline in the donor selection but none in the acceptor). But this choice is up you and you need to evaluate it based on the specific fitting approach you use.

**Comment by tritemio** [comment]
@ncodina, I'm closing this issue. Feel free to open a new issue if you have further questions.
